app/views/users/show.html.erb

```diff
@@ -1 +1,29 @@
-<h1>Here is a list of all posts from user 1 and bio</h1>
\ No newline at end of file
+<article class="main-article">
+  <div class="user-info-wrapper">
+    <%= image_tag @user.photo, :class => "user-avatar" %>
+    <div class="user-details">
+      <p><%= @user.name %></p>
+      <span>Number of posts: <%= @user.post_counter %></span>
+    </div>
+  </div>
+  
+  <div class="bio">
+    <h3>Bio</h3>
+    <p><%= @user.bio %></p>
+  </div>
+
+  <% @user.recent_posts.each_with_index do |post, index| %>
+    <div class="post">
+      <h4>Post #<%= index + 1 %></h4>
+      <div>
+        <p class="post-title"><%= post.title %></p>
+      </div>
+      <div class="post-footer">
+        <span>Comments: <%= post.comments_counter %></span>
+        <span>Likes: <%= post.likes_counter %></span>
+      </div>
+    </div>
+  <% end %>
```

Comment:
- [ ] [Optional] Nice work on displaying the post information! 👏 However, it would be beneficial to link each post to its show page. This way, clicking on each post would direct users to the respective show page. Adding these links will enhance the user experience and align with common navigation practices. Consider implementing this improvement to make your application more user-friendly. Keep up the good work! 🚀🔗